cpp/src/io/parquet/page_decode.cuh

```diff
@@ -1279,22 +1301,23 @@ inline __device__ bool setupLocalPageInfo(page_state_s* const s,
           // first 4 bytes are length of RLE data
           int const len = (cur[0]) + (cur[1] << 8) + (cur[2] << 16) + (cur[3] << 24);
           cur += 4;
-          if (cur + len > end) { s->error = 2; }
+          if (cur + len > end) { s->set_error_code(decode_error::RLE_STREAM_OVERRUN); }
```

Comment:
Here we're doing RLE encoded data. Perhaps it would be better to instead have the errors split up along level data vs actual data lines, so `LEVEL_STREAM_OVERRUN` and `DATA_STREAM_OVERRUN` vs the current `RLE_STREAM/STREAM_OVERRUN`

Reply:
Made the suggested changes, please check if I'm using the right codes in all places.

Reply:
Yep, I think it's good to go now.